CHANGES.md

```diff
@@ -1,8 +1,461 @@
+* enabled GitHub issues to file and track Chapel bugs and feature requests
+  (see TODO)
+* extended the Docker image to include a GASNet-based configuration
+  (see TODO)
+
+File/Directory Structure
+------------------------
+* moved object files for the compiler and runtime into $CHPL_HOME/build
+* changed the naming/structure of directories used to store binary files
+
+Configuration Changes
+---------------------
+* extended CHPL_LLVM to support new modes: llvm-minimal and system[-minimal]
+  (see TODO)
+
+Semantic Changes / Changes to Chapel Language
+---------------------------------------------
+* made functions return array expressions by value by default
+  (see TODO)
+* made the default intent for arrays be 'ref' if modified, else 'const ref'
+  (see TODO)
+* made the default intent for 'this' on record methods be 'ref' if modified
+  (see TODO)
+* made tuple semantics more closely follow the behavior of their element types
+  (see TODO)
+* deprecated support for the '=>' operator to create array aliases
+  (replace 'var A => B;' with 'ref A = B;' and
+   and 'var A: [D] => B;' with 'ref A = B.reindex(D);')
+* deprecated support for using '=>' in constructors to create alias fields
+
+Syntactic/Naming Changes
+------------------------
+* renamed the 'Error' module to 'SysError'
+  (see TODO)
+
+New Features
+------------
+* added prototype support for error-handling in Chapel
+* dramatically improved support for initializers
+* added support for deinit() as a replacement for class/record destructors
+* added support for 'forwarding' methods to fields of classes and records
+  (see TODO)
+* made 'void' a first-class type in the language and a way to fold variables
+  (see TODO)
+* added support for module deinit() functions
+
+Feature Improvements
+--------------------
+* vastly improved the stability, generality, and precision of 'where' clauses
+  (see TODO)
+* improved array slice, rank change, and reindexing support across domain maps
+* extended return intent overloading to select between 'const' and 'const ref'
+  (see TODO)
+* added support for using 'const' and 'const ref' as 'this' intents
+* added support for defining type aliases for generic classes
+  (e.g., 'type Vect = Vector;' for generic type 'Vector')
+* added the ability to define secondary methods on instantiated types
+  (e.g., 'proc (R(int)).foo() { ... }' defines foo on R(int) but not R(real))
+* added min() and max() overloads that accept and generate 'param' values
+* added support for casts between c_void_ptr and class objects/c_string
+* improved the default hash functions used by associative domains
+* reduced opportunities for races on set operations for associative arrays
+* enabled 'param's and 'config param's to be specified without initializers
+  (TODO: bradc document)
+
+Feature Slips
+-------------
+* domains of reindexed distributed arrays are not distributed as they should be
+* removed support for having first-class functions capture outer variables
```

Comment:
D'oh. For some reason I didn't read line 123. Not sure how I did that. Your bullet is good.